core/src/main/kotlin/org/evomaster/core/output/naming/ActionTestCaseNamingStrategy.kt

```diff
@@ -3,52 +3,56 @@ package org.evomaster.core.output.naming
 import com.webfuzzing.commons.faults.FaultCategory
 import org.evomaster.core.output.TestWriterUtils
 import org.evomaster.core.problem.enterprise.DetectedFaultUtils
-import org.evomaster.core.problem.httpws.HttpWsCallResult
-import org.evomaster.core.problem.rest.RestCallAction
 import org.evomaster.core.search.EvaluatedIndividual
 import org.evomaster.core.search.Solution
 import org.evomaster.core.search.action.EvaluatedAction
 
-open class ActionTestCaseNamingStrategy(
+abstract class ActionTestCaseNamingStrategy(
     solution: Solution<*>,
-    private val languageConventionFormatter: LanguageConventionFormatter
+    private val languageConventionFormatter: LanguageConventionFormatter,
 ) : NumberedTestCaseNamingStrategy(solution)  {
 
+    protected val on = "on"
+    protected val throws = "throws"
+    protected val returns = "returns"
+    protected val error = "error"
+    protected val success = "success"
+    protected val data = "data"
+    protected val empty = "empty"
 
-    override fun expandName(individual: EvaluatedIndividual<*>): String {
-        var evaluatedAction = individual.evaluatedMainActions().last()
-        var action = evaluatedAction.action as RestCallAction
-
-        return "_${languageConventionFormatter.formatName(listOf(action.verb.toString(), "on", getPath(action.path.nameQualifier), addResult(individual)))}"
+    protected fun formatName(nameTokens: MutableList<String>): String {
```

Comment:
unless `nameTokens` is supposed to be modified as side-effect of this function, it should be passed in input as `List` and not `MutableList` (note this latter is a subclass of `List`)